### ISSUE TITLE
🚀 Feature: Don't require version if package has `"private": true`

### ISSUE THREAD

**Issue by altano** [JoshuaKGoldberg/eslint-plugin-package-json#250]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I have disabled the `package-json/valid-version` rule because it warns on private packages. Tools like `changeset` require that you remove the private package version number (instead of, say providing a useless one like 0.0.1) or it will try to manage that version, even if the package is private. Ideally this package would just allow no versions on private packages.

Thank you for making this package btw. It's great!

**Comment by JoshuaKGoldberg** [comment]
Oh, that's good to know - thank you for reporting!

Right now, the rule's implementation is set to only complain if there is a `"version"` property. It doesn't complain if the property doesn't exist:

https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/blob/2ce188156d0b91fdcb9606e069a45148c673e6c0/src/tests/rules/valid-version.test.ts#L45-L46

What situation are you seeing where there's an erroneous complaint?

**Comment by altano** [comment]
Sorry, I misspoke: it's the `valid-package-def` rule that is erroring when a package has a private specifier and no version:

```
/Volumes/sourcecode/npm-packages/packages/eslint-config-altano/package.json
  1:1  error  Missing required field: version  package-json/valid-package-def
```

This test case in valid-package-def.test.ts would reveal it:

```
		{
			code: `{
  "name": "panda",
  "private": true
}`,
			filename: "package.json",
		},
```

I can submit a PR if you'd like. I would create an exception for this error and only pass the `Missing required field: version` error through if `private: true` is not in the package? (I don't see a way of configuring package validator to use that logic itself). Would I somehow traverse the ast to determine it has a name, or ?

**Comment by JoshuaKGoldberg** [comment]
> I don't see a way of configuring package validator to use that logic itself

Ahh, good news, I'm also working on https://github.com/TechNickAI/package.json-validator. So this would be a good issue to file there!

**Comment by JoshuaKGoldberg** [comment]
Ha, just hit this in my own work.

Clarifying my request for the issue: I like giving authors of issues & PRs proper attribution, and having them file things means they get credit more prominently. But I don't want to add _more_ work to your plate on this clear bug in libraries that's inconveniencing you already. I've set myself a reminder to do everything on Monday if nothing's filed by then.
